src/core/api/replication_execution.go

```diff
@@ -146,68 +188,21 @@ func (r *ReplicationOperationAPI) CreateExecution() {
 
 // GetExecution gets one execution of the replication
 func (r *ReplicationOperationAPI) GetExecution() {
-	executionID, err := r.GetInt64FromPath(":id")
-	if err != nil || executionID <= 0 {
-		r.SendBadRequestError(errors.New("invalid execution ID"))
-		return
-	}
-	execution, err := replication.OperationCtl.GetExecution(executionID)
-	if err != nil {
-		r.SendInternalServerError(fmt.Errorf("failed to get execution %d: %v", executionID, err))
-		return
-	}
-
-	if execution == nil {
-		r.SendNotFoundError(fmt.Errorf("execution %d not found", executionID))
-		return
-	}
-	r.WriteJSONData(execution)
+	r.WriteJSONData(r.execution)
 }
 
 // StopExecution stops one execution of the replication
 func (r *ReplicationOperationAPI) StopExecution() {
-	executionID, err := r.GetInt64FromPath(":id")
-	if err != nil || executionID <= 0 {
-		r.SendBadRequestError(errors.New("invalid execution ID"))
-		return
-	}
-	execution, err := replication.OperationCtl.GetExecution(executionID)
-	if err != nil {
-		r.SendInternalServerError(fmt.Errorf("failed to get execution %d: %v", executionID, err))
-		return
-	}
-
-	if execution == nil {
-		r.SendNotFoundError(fmt.Errorf("execution %d not found", executionID))
-		return
-	}
-
-	if err := replication.OperationCtl.StopReplication(executionID); err != nil {
-		r.SendInternalServerError(fmt.Errorf("failed to stop execution %d: %v", executionID, err))
+	if err := replication.OperationCtl.StopReplication(r.execution.ID); err != nil {
```

Comment:
Should check the r.execution is nil to avoid nil pointer

Reply:
It is checked in the prepare